data/extract_rosbag_to_tf.py

```diff
@@ -177,12 +192,13 @@ def main():
     else:
         t_end = t_start + args.end_time
 
+    eps = 0.1
     for topic, msg, t in bag.read_messages(
             topics=['/davis/left/image_raw',
                     '/davis/right/image_raw',
                     '/davis/left/events',
                     '/davis/right/events'],
-            start_time=rospy.Time(args.start_time + t_start),
+            start_time=rospy.Time(max(args.start_time, eps) - eps + t_start),
```

Comment:
It seems like this will always skip the first 0.1s of the bag if start_time is 0. Is this intentional?

Reply:
It does not skip 0.1s of the bag if start_time is 0.
The proposed condition is equal to:
'''
if args.start_time < eps:
    start_time = t_start
else:
    start_time = t_start + args.start_time - eps
'''
It solves the issue with testing on outdoor_day1 with self.start_time=222.4. when I tested using bag files and .npy, the first event count_image's were different. It is due to difference in timestamps of message and stored events.

Reply:
Oh I see, I guess what happened was that there isn't a grayscale frame for a little bit of time at 222.4, but there is one within 0.1s before this time? What should happen in the code is that events are not collected until after the first grayscale frame, so data would be extracted from the first grayscale timestamp after 222.4. Did you see a noticeable difference with this fix?

Reply:
You are right originally you do not collect events before the first grayscale frame.
And there is a frame at 222.4, but there are also events in a message with timestamp less than 222.4, but their stored timestamps are great than 222.4. For example, this code can show you the difference.
```
event_mv=0
for topic, msg, t in bag.read_messages(
    topics=['/davis/left/image_raw',
            '/davis/right/image_raw',
            '/davis/left/events',
            '/davis/right/events'],
            start_time=rospy.Time(t_start),
            end_time=rospy.Time(t_end)):
    if  'events' in topic and msg.events:
        for event in msg.events:
            ts = event.ts # <---------------------------------- sometimes ts > t.
            event_mv = max(event_mv, (ts - t).to_sec())
print(event_mv)
```
I have no idea why it is so. I'm just trying to get all events that are between grayscale frames according to the stored timestamps (ts)